src/number_theory/zeta_function.lean

```diff
@@ -0,0 +1,373 @@
+/-
+Copyright (c) 2023 David Loeffler. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Loeffler
+-/
+import analysis.mellin_transform
+import number_theory.modular_forms.jacobi_theta
+
+/-!
+# Definition of the Riemann zeta function
+
+## Main definitions:
+
+* `riemann_zeta`: the Riemann zeta function `ζ : ℂ → ℂ`.
+* `riemann_completed_zeta`: the completed zeta function `Λ : ℂ → ℂ`, which satisfies
+  `Λ(s) = π ^ (-s / 2) Γ(s / 2) ζ(s)` (away from the poles of `Γ(s / 2)`).
+* `riemann_completed_zeta₀`: the entire function `Λ₀` satisfying
+  `Λ₀(s) = Λ(s) + 1 / (s - 1) - 1 / s` wherever the RHS is defined.
+
+Note that mathematically `ζ(s)` is undefined at `s = 1`, while `Λ(s)` is undefined at both `s = 0`
+and `s = 1`. Our construction assigns some values at these points (which are not arbitrary, but
+I haven't checked exactly what they are).
```

Comment:
I would just remove the parenthesis (and maybe check what the values are with some simp lemmas, they shouldn't be hard to compute). Hopefully when you have a meromorphic function the value we assign at the function at the pole should be the constant coefficient of the Laurent expansion.

Reply:
> and maybe check what the values are with some simp lemmas, they shouldn't be hard to compute

You can take my word as a number theorist: every property of the Riemann zeta function is fascinating, beautiful, and furiously hard! 😄

You are correct that these definitions assign `Λ(1)` to be the constant coefficient of the Laurent series of `Λ(s)` around `s = 1`. But computing this constant coefficient is definitely not trivial! There is an explicit formula involving the Euler–Mascheroni constant, but it would take some serious effort to prove it.

Reply:
It only takes 20 lines or so to prove
```lean
lemma riemann_completed_zeta_one : 
  riemann_completed_zeta 1 = lim (𝓝[≠] (1 : ℂ)) (λ s, riemann_completed_zeta s - 1 / (s - 1))
```
and the analogue at s = 1. This at least shows how to connect the values we assign at the poles to something which is mathematically natural! Shall I add it to this PR?

Reply:
(The actual value `riemann_completed_zeta 1` appears to be $(\gamma - \log 4\pi) / 2$.)

Reply:
This can definitely wait for another PR.